File: visualize/try_except.py
import sys

def process_data(data_list):
    if not data_list:
        print("List is empty.")
        return 0

    total = 0
    for item in data_list:
        try:
            # エラーノード1: 型変換
            value = int(item)
            # ここから if 分岐まで（分析対象区間）
            if value % 2 == 0:
                print(f"{value} is even.")
                total += value
            else:
                print(f"{value} is odd.")
            try:
                result = value / (value % 3 or 1)
                if result > 10:
                    print(f"Result is large: {result}")
                    total += result
                else:
                    print(f"Result is small: {result}")
            except ZeroDivisionError:
                print(f"Division error for item: {item}", file=sys.stderr)
                continue
        except (ValueError, TypeError):
            print(f"Skipping invalid item: {item}", file=sys.stderr)
            continue
        except ZeroDivisionError:
            print(f"Division error for item: {item}", file=sys.stderr)
            continue
        except KeyError:
            print(f"Key error for item: {item}", file=sys.stderr)
            continue

    print("Processing complete.")
    return total

# シンプルなテストデータ
numbers = ["1", "2", "3", "four", "5", "6", "0"]
result = process_data(numbers)
print("Total sum of even numbers:", result)

File: visualize/try_except.java
import java.io.PrintStream;
import java.util.List;
import java.util.ArrayList;

public class DataProcessor {

    public static int processData(List<String> dataList) {
        if (dataList == null || dataList.isEmpty()) {
            System.out.println("List is empty.");
            return 0;
        }

        int total = 0;
        for (String item : dataList) {
            try {
                int value = Integer.parseInt(item);
                if (value % 2 == 0) {
                    System.out.println(value + " is even.");
                    total += value;
                } else {
                    System.out.println(value + " is odd.");
                }
            } catch (NumberFormatException e) {
                // 'four' のような無効な文字列を処理する
                System.err.println("Skipping invalid item: " + item);
                continue; // Javaのcontinueと同様
            }
        }

        System.out.println("Processing complete.");
        return total;
    }

    public static void main(String[] args) {
        List<String> numbers = new ArrayList<>();
        numbers.add("1");
        numbers.add("2");
        numbers.add("3");
        numbers.add("four");
        numbers.add("5");
        numbers.add("6");

        int result = processData(numbers);
        System.out.println("Total sum of even numbers: " + result);
    }
}
